src/main/kotlin/com/openlattice/entitysets/controllers/EntitySetsController.kt

```diff
@@ -189,9 +191,65 @@ constructor(
                 .map { AuthorizationUtils.getLastAclKeySafely(it) }
                 .toSet()
 
-        return  entitySetManager.getEntitySetsAsMap(entitySetIds).values.toSet()
+        return entitySetManager.getEntitySetsAsMap(entitySetIds).values.toSet()
     }
 
+    @Timed
+    @RequestMapping(
+            path = [BY_ID_PATH],
+            method = [RequestMethod.POST],
+            produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getEntitySetsById(entitySetIds: Set<UUID>): Map<UUID, EntitySet> {
+
+        entitySetIds.forEach { entitySetId -> ensureReadAccess(AclKey(entitySetId)) }
+
+        val entitySets = entitySetManager.getEntitySetsAsMap(entitySetIds)
+        val now = OffsetDateTime.now()
+        entitySets.forEach { (entitySetId: UUID) -> recordEvent(
+                AuditableEvent(
+                        getCurrentUserId(),
+                        AclKey(entitySetId),
+                        AuditEventType.READ_ENTITY_SET,
+                        "EntitySet read through EntitySetsApi.getEntitySetsById",
+                        Optional.empty(),
+                        ImmutableMap.of(),
+                        now,
+                        Optional.empty()
+                )
+        )}
+
+        return entitySets
+    }
+
+    @Timed
+    @RequestMapping(
+            path = [BY_NAME_PATH],
+            method = [RequestMethod.POST],
+            produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getEntitySetsByName(entitySetNames: Set<String>): Map<String, EntitySet> {
+
+        val entitySetIds = edmManager.getAclKeyIds(entitySetNames).values.toSet()
+        entitySetIds.forEach { entitySetId -> ensureReadAccess(AclKey(entitySetId)) }
```

Comment:
This could be also bulked by using `accessCheck( Map<AclKey, EnumSet<Permission>> requiredPermissionsByAclKey )`